include/nuttx/fs/ioctl.h

```diff
@@ -405,15 +405,33 @@
 
 /* FIFOs and pipe driver ioctl definitions **********************************/
 
-#define _PIPEIOCVALID(c)  (_IOC_TYPE(c)==_PIPEBASE)
-#define _PIPEIOC(nr)      _IOC(_PIPEBASE,nr)
-
-#define PIPEIOC_POLICY    _PIPEIOC(0x0001)  /* Set buffer policy
-                                             * IN: unsigned long integer
-                                             *     0=free on last close
-                                             *       (default)
-                                             *     1=fre when empty
-                                             * OUT: None */
+#define _PIPEIOCVALID(c)    (_IOC_TYPE(c)==_PIPEBASE)
+#define _PIPEIOC(nr)        _IOC(_PIPEBASE,nr)
+
+#define PIPEIOC_POLICY      _PIPEIOC(0x0001)  /* Set buffer policy
+                                               * IN: unsigned long integer
+                                               *     0=free on last close
+                                               *       (default)
+                                               *     1=fre when empty
+                                               * OUT: None */
+
+#define PIPEIOC_POLLINTHRD  _PIPEIOC(0x0002)  /* Set pipe POLLIN
+                                               * notifty buffer threshold.
+                                               * IN: unsigned long integer.
+                                               *     POLLIN only occurs when
+                                               *     buffer contains more
```

Comment:
should we use >= for PIPEIOC_POLLINTHRD  and PIPEIOC_POLLOUTTHRD and change the default valu of d_polloutthrd  and d_pollinthrd to 1?

Reply:
I think > for POLLIN/POLLOUT threshold is more reasonable and readable.  For example, in local udp socket, we should avoid triggering POLLIN when buffered content is the size of preamble, that is "more than" logic in its nature.

Though > and >= both can be implemented, and I can replace it with >= and change default value to 1. I don't think ">="  is more readable.

Reply:
Ok